simphony_ui/couple_CFDDEM.py

```diff
@@ -189,40 +231,17 @@ def run_calc(output_path, mesh_name):
     liggghts_wrapper.SP_extension[liggghts.CUBAExtension.PAIR_POTENTIALS] = \
         ['repulsion', 'cohesion']
 
-    if openfoam_mode is not "none":
-        # Generate cell list
-        cellmat = {}
-        index = {}
-        gridsize = [openfoam_chansize[0] / openfoam_numgrid[0],
-                    openfoam_chansize[1] / openfoam_numgrid[1],
-                    openfoam_chansize[2] / openfoam_numgrid[2]]
-
-        for cell in openfoam_mesh.iter_cells():
-            lln = [
-                openfoam_chansize[0] * 2,
-                openfoam_chansize[1] * 2,
-                openfoam_chansize[2] * 2]
-            for k in range(0, 8):
-                for i in range(0, 3):
-                    if openfoam_mesh.get_point(
-                            cell.points[k]).coordinates[i] < lln[i]:
-                        lln[i] = openfoam_mesh.get_point(
-                            cell.points[k]).coordinates[i]
-
-            for i in range(0, 3):
-                index[i] = round(lln[i]/gridsize[i])
-
-            cellmat[index[0], index[1], index[2]] = cell.uid
+    cellmat = openfoam_model.generate_cell_list()
```

Comment:
just cell_list(). While it's true you are generating it, you are basically getting information from an object. The fact that it computes it is irrelevant from the outside.